test/mpi/test_barrier.cc

```diff
@@ -77,6 +77,10 @@ void TestBarrier::run()
         if  (0 == rank && !completed) {
             MPI_Wait(&rreq, MPI_STATUS_IGNORE);
         }
+        if (i < size - 1) {
```

Comment:
why not for the last rank?

Reply:
if this is the last iteration (i == size - 1) then i don't need to re-init for next iteration - there is no next iteration. coll_finalize is called in destructor.